### PR TITLE
Move cert-manager to the github.com/cert-manager organization

### PR DESCRIPTION
**Pull request by wallrj**
* Updated references to jetstack/cert-manager
* Updated references to `org: jetstack` for cert-manager related tests and config
* Remove unnecessary repo specific config which is now inherited from cert-manager  organization
* Removed some lingering references to jetstack/cert-manager-webhook-example, which has already been moved to cert-manager org.

Based on  @jakexks 's original PR:
 * https://github.com/jetstack/testing/pull/442

### CODE REVIEW

File: config/config.yaml
Comment (munnerz) [inline config/config.yaml:34]: Can we not do this now already? I'd have thought `cert-manager` as an org should definitely have it enabled org wide (it wasn't with jetstack due to various internal projects/things that didn't use Prow)

File: config/config.yaml
Comment (munnerz) [inline config/config.yaml:66]: This whole `jetstack` entry can be removed now right?

File: config/config.yaml
Comment (munnerz) [inline config/config.yaml:295]: Have we not been freezing merges during a codefreeze near release? If so, +1 to this going. But this was here so that we could block merges unless they were explicitly in the milestone we are soon to be releasing.

(i.e. near 1.7 release, PRs *must* have the `v1.7` milestone on them to be merged to master). The idea being to make it possible to lgtm/approve stuff despite it being targeted to land in the `v1.8` release.

File: config/config.yaml
Comment (munnerz) [inline config/config.yaml:311]: Manually reviewed the 'diff' here and it LGTM :)

File: config/jobs/cert-manager/cert-manager-periodics.yaml
Comment (munnerz) [inline config/jobs/cert-manager/cert-manager-periodics.yaml:8]: would it make sense to move all these test jobs into `config/jobs/cert-manager/cert-manager/...`? Then we have `config/jobs/{org}/{repo}` and we're consistent everywhere :)

File: config/plugins.yaml
Comment (munnerz) [inline config/plugins.yaml:188]: 👍 just confirming that this diff on plugins LGTM

File: legacy/README.md
Comment (munnerz) [inline legacy/README.md:20]: 👍 although a follow up may be to remove this legacy stuff :) 🎉  pretty sure bootstrap.py is no longer used anywhere

File: config/config.yaml
Comment (wallrj) [reply config/config.yaml]: Done.

File: config/config.yaml
Comment (wallrj) [reply config/config.yaml]: Done.

File: config/config.yaml
Comment (wallrj) [reply config/config.yaml]: Done...I think...I'm not actually sure where to put that stanza. After 3 attempts I've put it beneath the cert-manager org so that it only affects repos in that org. 
Lets see if the linter allows this.

File: config/config.yaml
Comment (wallrj) [reply config/config.yaml:256]: It seems to me that Tide turns these in to a series of GitHub search queries and that a given PR has to appear in all the search results before it is merged.

https://github.com/kubernetes/test-infra/blob/beadb8f487e14fb2f5145758590df4c8357d08c9/prow/config/tide.go#L172-L175

To be on the safe side, I'll duplicate all the config into this query too.

File: config/config.yaml
Comment (wallrj) [reply config/config.yaml:295]: That's not the documented process anyway. https://cert-manager.io/docs/contributing/release-process/#process-for-releasing-a-version

We fast-forward the release-x.y.z branch for alpha releases  and for the first beta release
and then do cherry-picks after that.

What's the use-case for / advantage of the milestone check?
I guess if there's a feature  which is definitely not to be in the next release and that is being merged in parts and partly merged even before the first beta, then that should be merged not into master, but into a feature-branch which can then be merged once the feature is complete.

File: config/config.yaml
Comment (wallrj) [reply config/config.yaml:272]: Done.

File: config/jobs/cert-manager/cert-manager-periodics.yaml
Comment (wallrj) [reply config/jobs/cert-manager/cert-manager-periodics.yaml:8]: Done.

File: config/config.yaml
Comment (munnerz) [reply config/config.yaml:295]: The idea stems from the "code freeze" concept in Kubernetes where after a certain point in the release (I believe when the first beta is cut), "milestone maintainers" are the only ones that can use the `/milestone` command to add a milestone to PRs.

The master branch begins requiring that PRs into master *must* be explicitly approved for the upcoming release - so anything that isn't approved won't actually merge until the code freeze is lifted (which I believe in k8s is when the first *RC* of the next release is cut).

This means:

* during the 'alpha' phase (i.e. immediately after release N-1), PRs targeting master must have `lgtm` and `approved` only (like we do today)
* during the 'beta' phase (i.e. immediately after the first *beta* of release N), PRs targeting master must have `lgtm`, `approved` *and* target milestone `N`. We then continue to *fast forward* the release branch to the tip of HEAD. The idea is to encourage PRs that focus on stability of the upcoming release (or in some instances, feature PRs that have been excepted as they are almost ready and just need a few extra fixes/deemed to be critical and there's been a discussion to allow it).
* during the 'rc' phase (i.e. immediately after the first *RC* release of N), PRs must be *cherry picked* into the appropriate release branch as 'master' is now tracking release N+1.

After the stable 'N' is cut, not much changes from the RC phase except things targeting the release branch are now going to land in `vMAJOR.N.1` (i.e. the first *patch* release of release N after the initial stable).

We don't do release candidates (RCs) so that step hasn't really ever been done, but we've enforced this for a handful of releases in the past (though not recently!)

**Review by munnerz — APPROVED** [review]

**Pull request merged**
